### PR TITLE
NMS-13645 Smoke tests should use HealthCheck Rest instead of connecting to SSH

### PR DESCRIPTION
**Pull request by aramosvizcarra**
Updated ssh Health Check with Rest Health check for Minion and Sentinel.

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13645

### CODE REVIEW

File: smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java [hunk 1]
```diff
@@ -241,8 +247,17 @@ public WaitForMinion(MinionContainer container) {
         @Override
         protected void waitUntilReady() {
             LOG.info("Waiting for Minion health check...");
-            final InetSocketAddress sshAddr = container.getSshAddress();
-            awaitHealthCheckSucceeded(sshAddr, 5, "Minion");
```
Comment (swachter) [inline smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java:245]: Do we still need the code for `awaitHealthCheckSucceeded`?

**Review by swachter — COMMENTED** [review]
Looks great! Only one comment.

File: smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java (hunk 1 above)
Comment (aramosvizcarra) [reply smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java:245]: That's a good question.  @cgorantla what do you think?.

Comment (cgorantla) [reply smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java:245]: May be it  will be used  in another smoke test which should still check for SSH connection on Minion/Sentinel.

**Review by cgorantla — APPROVED** [review]

File: smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java (hunk 1 above)
Comment (swachter) [reply smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java:245]: * I propose that all integration test now rely on the rest api when waiting for container startup.
* There should be an additional integration test though, that checks if the `health-check` command is available on the command line. That test could be based on the code of the current ssh-based startup check.

Comment (aramosvizcarra) [reply smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java:245]: Yes I agree. I'll include that in this PR

**Review by swachter — APPROVED** [review]
LGTM - Thanks!

**Review by cgorantla — APPROVED** [review]

**Pull request merged**